hw1403recyclerview/src/main/java/com/tashad16a/gmail/hw1403recyclerview/recyclerview/RecyclerViewAdapter.kt

```diff
@@ -0,0 +1,81 @@
+package com.tashad16a.gmail.hw1403recyclerview.recyclerview
+
+import android.graphics.Color
+import android.support.annotation.NonNull
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import com.tashad16a.gmail.hw1403recyclerview.itemtouch.ItemTouchHelperAdapter
+import com.tashad16a.gmail.hw1403recyclerview.R
+import java.util.Collections
+
+class RecyclerViewAdapter : RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>,
+    ItemTouchHelperAdapter {
+
+    var students: MutableList<Student> = ArrayList()
+    lateinit var studentView: View
+
+    constructor(students: MutableList<Student>) : super() {
+        this.students = students
+    }
+
+    override fun onCreateViewHolder(@NonNull viewGroup: ViewGroup, viewType: Int): ViewHolder {
+        studentView = LayoutInflater.from(viewGroup.context).inflate(R.layout.layout_student, viewGroup, false)
```

Comment:
onCreateViewHolder called once for each type of view.
onBindViewHolder called every time when in RecyclerView redrawing some view.
So, save inflated view into studentView field and use it in onItemSelected() method is wrong